File: javascript01/step04/test02.js
/* 주제:함수 - 정의와 사용 II
*/
"use strict"
// - 1) 파라미터 없는 함수
function f1(){
  console.log("f1() 호출 됨");
}

f1();
// - 2) 파라미터 x 리턴 값o
function f2(){
  return "hello";
}
var value = f2();
console.log("f2의 리턴 값은= "+ value);
// - 3) 파라미터 o 리턴 값 x
function f3(a){
  console.log("--->"+a);
}

f3(100);
f3(200);
f3(300);
// - 4) 파라미터 o 리턴 값 o
function f4(a){
  return "a = "+ a + "입니다";
}

var value = f4(100);
console.log(value);

// 굳이 값을 별도의 변수에 저장한 후 출력할 필요는 없다
// 함수의 리턴 값을 즉시 사용할 수 있다.
// 참고! 실행은 항상 괄호() 안쪽부터 실행한다.

console.log(f4(200));

File: java01/src/step05/Test02.java
package step05;
/* 메서드 : 메서드를 정의하는 문법 */
public class Test02 {
  //1) 값을 리턴하지 않고, 파라미터를 받지 않는다.
  //=> 값을 리턴하지 않을 때는 반드시 메서드 선언 앞에 void를 붙여야 한다.
  //=> 아규먼트를 받지 않는다면 빈 채로 두어야 한다.
  public static void m1() {
    System.out.println("m1()");
  }
  
  //2) 값을 리턴하지 않고, 아규먼트는 받는다.
  //=> 아규먼트 (argument)? 함수를 호출 할 때 넘겨주는 값
  //=> 파라미터(parameter)? 메서드가 호출될 때 넘어온 값을 받는 로컬 변수
  //    그러나 현장에서는 두 용어를 구분하지 않고 섞어 쓴다.
  public static void m2(int a, String str, boolean b) {
    System.out.printf("%d, %s, %b\n", a, str, b);
  }
  
  public static String m3() {
    return "Hello";
    //System.out.println("안되")//안된다;
  }
  //3) 값을 리턴하고, 아규먼트를 받지 않는다.
  //=> 따라서 reutrn 문장 다음에 명령문을 작성해서는 안된다.
  
  
//4) 값을 리턴하고, 아규먼트를 받는다.
  //=> 보통 값을 받아서 처리한 후 그 결과를 리턴하고자 할 때 사용한다.
  public static int m4(int a) {
    return a * a;
  }
  public static void main(String[] args) {
    m1();
    m2(10, "홍길동", false);
    String value = m3();
    System.out.println(value);
    int result = m4(16);
    System.out.println(result);
  }
  
  
}
